core/src/mindustry/logic/BuildingStringBuilder.java

```diff
@@ -0,0 +1,51 @@
+package mindustry.logic;
+
+import arc.util.*;
+import arc.math.geom.*;
+import mindustry.*;
+import mindustry.gen.*;
+import mindustry.world.blocks.logic.LogicBlock;
+import mindustry.world.blocks.logic.MessageBlock;
+
+public class BuildingStringBuilder implements CharSequence{
+    public Point2 source;
+
+    public BuildingStringBuilder(Point2 source){
+        this.source = source;
+    }
+
+    public @Nullable StringBuilder obtainBuilder(){
+        @Nullable Building src = Vars.world.build(source.x, source.y);
+        if(src == null)return(null);
```

Comment:
1. You're using `return(value)` *everywhere* across this PR, which is incorrect style. `return` is not a method call, it should not need parentheses. Java uses `return value;`
2. You're not adding a space after the `)`

I'm highlighting the issue for this particular line, but it's present all over the PR. Please re-read the style guidelines/look at the format of existing code.